consumer/sm_context.go

```diff
@@ -456,27 +479,39 @@ func SendUpdateSmContextHandoverBetweenAMF(
 			}
 		}
 	}
-	return SendUpdateSmContextRequest(smContext, updateData, nil, nil)
+	return SendUpdateSmContextRequest(ctx, smContext, updateData, nil, nil)
 }
 
-func SendUpdateSmContextRequest(smContext *amf_context.SmContext,
+func SendUpdateSmContextRequest(ctx context.Context, smContext *amf_context.SmContext,
 	updateData models.SmContextUpdateData, n1Msg []byte, n2Info []byte) (
 	response *models.UpdateSmContextResponse, errorResponse *models.UpdateSmContextErrorResponse,
 	problemDetail *models.ProblemDetails, err1 error,
 ) {
+	ctx, span := tracer.Start(ctx, "HTTP PUT smf/sm-contexts/{smContextRef}")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("http.method", "PUT"),
+		attribute.String("nf.target", "smf"),
+		attribute.String("net.peer.name", smContext.SmfUri()),
+		attribute.String("amf.nf.id", amf_context.AMF_Self().NfId),
+		attribute.String("smf.nf.id", smContext.SmfID()),
+		attribute.String("smf.uri", smContext.SmfUri()),
+	)
+
 	configuration := Nsmf_PDUSession.NewConfiguration()
 	configuration.SetBasePath(smContext.SmfUri())
 	client := Nsmf_PDUSession.NewAPIClient(configuration)
 
-	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
Why did you change name of the variable to `timeoutCtx` while in other places (in this file and in other files) you keep it as `ctx` (i.e., `ctx, cancel := context.WithTimeout(ctx, 30*time.Second)`)?